allennlp/common/from_params.py

```diff
@@ -116,6 +116,37 @@ def create_kwargs(cls: Type[T], params: Params, **extras) -> Dict[str, Any]:
     params.assert_empty(cls.__name__)
     return kwargs
 
+
+def create_extras(cls: Type[T],
+                  extras: Dict[str, Any]) -> Dict[str, Any]:
+    """
+    Given a dictionary of extra arguments, returns a dictionary of
+    kwargs that actually are a part of the signature of the cls.from_params
+    (or cls) method.
+    """
+    subextras: Dict[str, Any] = {}
+    if hasattr(cls, "from_params"):
+        from_params_method = cls.from_params
+    else:
+        # In some rare cases, we get a registered subclass that does _not_ have a
+        # from_params method (this happens with Activations, for instance, where we
+        # register pytorch modules directly).  This is a bit of a hack to make those work,
+        # instead of adding a `from_params` method for them somehow. Then the extras
+        # in the class constructor are what we are looking for, to pass on.
+        from_params_method = cls
+    if takes_arg(from_params_method, "extras"):
```

Comment:
I think we actually want a check for something like `**extras`, whatever its name, instead of just hard-coding the _name_ `extras`.  This possible by checking for the `kind` of a parameter with `inspect`:

```python
import inspect

def f(a: str, *inputs, **kwargs):
    pass

s = inspect.signature(f)
for p in s.parameters.values():
    print(p.name, p.kind)
```

output:
```
a POSITIONAL_OR_KEYWORD
inputs VAR_POSITIONAL  
extras VAR_KEYWORD     
```

So, a function like `takes_kwargs(method)` that checks for the presence of a `VAR_KEYWORD` argument would be better here.

Reply:
Looking at the original code, it looks like you're inheriting this logic instead of adding it, but it'd still be good to fix it while we're touching this code.

Reply:
The latest commit fixes this issue.